app/demo-geant-integration/SensitiveDetector.cc

```diff
@@ -41,9 +54,29 @@ G4bool SensitiveDetector::ProcessHits(G4Step* step, G4TouchableHistory*)
         return false;
     }
 
-    auto time = step->GetPreStepPoint()->GetGlobalTime();
+    // Fill the hit data for this step
+    auto         touchable = step->GetPreStepPoint()->GetTouchable();
+    unsigned int id        = touchable->GetVolume()->GetCopyNo();
+    HitData      data{id,
+                 edep,
+                 step->GetPreStepPoint()->GetGlobalTime(),
+                 touchable->GetTranslation()};
+
+    // Add energy deposition for this cell if it was hit before
+    for (auto hit : *(collection_->GetVector()))
```

Comment:
In this case using the type definition makes the code easier to read
```suggestion
    for (SensitiveHit* hit : *(collection_->GetVector()))
```

Reply:
This part of codes are dropped now.  Good comment though!